source/codegen/templates/service.cpp.mako

```diff
@@ -44,6 +44,26 @@ namespace ${config["namespace_component"]}_grpc {
       output->Add(item != VI_FALSE);
     }
   }
+
+% endif
+% if common_helpers.has_enum_array_string_out_param(functions):
+  template <typename T1, typename T2>
+  void ${service_class_prefix}Service::CopyEnumValues(const T1* input, T2* output, int length, std::map<T1, std::int32_t> enum_map)
+  {
+    for(int i = 0; i < length; i++)
```

Comment:
You should use interators instead of a for loop indexed by `i`. See the other copy functions as an example.

Also add spacing before the paren

Reply:
This loop was added to generalize this function for int32 and other arrays if they need it in future. The C arrays don't work with the iterator loops right?

Reply:
When do you foresee us needing to transfer from anything other than a vector to a RepeatedField? Be consistent with what we've got and don't generalize for scenarios that don't exist yet.

Reply:
changed this